crates/sui-core/src/consensus_adapter.rs

```diff
@@ -402,6 +396,35 @@ impl ConsensusAdapter {
         )
     }
 
+    fn await_submit_delay_user_transaction(
+        &self,
+        committee: &Committee,
+        tx_digest: &TransactionDigest,
+    ) -> (Duration, usize, bool) {
+        let (mut position, mapped_to_low_scoring) = self.submission_position(committee, tx_digest);
+
+        const MAX_LATENCY: Duration = Duration::from_secs(5 * 60);
+        const DEFAULT_LATENCY: Duration = Duration::from_secs(5);
```

Comment:
An alternative is to use `submit_delay_step` as the non-default value of DEFAULT_LATENCY or MAX_LATENCY.

Reply:
But then it wouldn't be a constant delay, right ? I wanted to have a constant delay step and also leave this logic here untouched even for reporting/metrics reasons as well so it can keep calculating latencies.